src/app/Fake.Core.ReleaseNotes/ReleaseNotes.fs

```diff
@@ -66,16 +66,40 @@ type ReleaseNotes =
     static member New(assemblyVersion,nugetVersion,notes) = ReleaseNotes.New(assemblyVersion,nugetVersion,None,notes)
 
 let private parseVersions =
-    let nugetRegex = String.getRegEx @"([0-9]+.)+[0-9]+(-[a-zA-Z]+\d*)?(.[0-9]+)?"
+    // https://github.com/fsprojects/FAKE/issues/2557
+    let nugetRegexLegacy = String.getRegEx @"([0-9]+.)+[0-9]+(-[a-zA-Z]+\d*)?(.[0-9]+)?"
+    let nugetRegex = 
+        /// From Fake.Core.SemVer
+        let pattern =
+            @"^(?<major>\d+)" +
+            @"(\.(?<minor>\d+))?" +
+            @"(\.(?<patch>\d+))?" +
+            @"(\-(?<pre>[0-9A-Za-z\-\.]+))?" +
+            @"(\+(?<build>[0-9A-Za-z\-\.]+))?$"
+        String.getRegEx pattern
     let assemblyVersionRegex = String.getRegEx @"([0-9]+.)+[0-9]+"
     fun line ->
         let assemblyVersion = assemblyVersionRegex.Match line
         if not assemblyVersion.Success
         then failwithf "Unable to parse valid Assembly version from release notes (%s)." line
 
-        let nugetVersion = nugetRegex.Match line
-        if not nugetVersion.Success
-        then failwithf "Unable to parse valid NuGet version from release notes (%s)." line
+        let nugetVersion = 
```

Comment:
Can we try to decrease the nesting level in this code block? Maybe try to replace if..else with `Option` API or match expression? Or do u have any other suggestions?

Reply:
This should decrease nesting, but evaluates `nugetRegexLegacy.Match line` even if not necessary.

```fsharp
  let nugetVersion = 
      /// Must split by whitespace to try match start of line and end of line in SemVer regex pattern
      let nugetVersion = 
          line.Split(' ') 
          |> Array.tryPick (fun segment -> 
            let m = segment.Trim() |> nugetRegex.Match
            if m.Success then Some m else None
          )
      // Add support for "nugetRegexLegacy" after change to correct SemVer parsing.
      // This should lead to the least disruption to users.
      let nugetVersionLegacy = nugetRegexLegacy.Match line
      // if isSome then it must be Success, so no need to check for that
      match nugetVersion, nugetVersionLegacy.Success with
      | Some nugetVersionValue, _ -> nugetVersionValue
      | None, true -> nugetVersionLegacy
      | None, false ->
          failwithf "Unable to parse valid Nuget version from release notes (%s)." line
```

Reply:
I agree, but this is more readable than if..else version. So I think readability wins here.
Also, can u convert the legacy value to an option? So that in match expression u will be matching on option values and not booleans and options.

Thanks

Reply:
Yeah matching on option and bool was bugging me too. Cleaned it up, will push after running tests.